File: 2nd round/weird-order/weird-order.py
def tray(c):
    return l[ord(c)-97]
s=input()
l=[0 for i in range(26)]
for i in range(26):
    l[ord(s[i])-97]=i
t=int(input())
while(t):
    t-=1
    s=input()
    m=list(s)
    m.sort(key=tray)
    for c in m:
        print(c,end='')
    print()

File: 2nd round/weird-order/weird-order.cpp
#include<bits/stdc++.h>
using namespace std;
int a[26];
bool tray(char x,char b)
{
    return a[x-97]<a[b-97];
}
int main()
{
    int i,j,k,x,y,z;
    char b;
    for(i=0;i<26;i++)
{
cin>>b;
a[b-97]=i;
}
int t;
cin>>t;
while(t--)
{
    string s;
    cin>>s;
    sort(s.begin(),s.end(),tray);
    cout<<s<<"\n";
}
    return 0;

}